jbmc/src/java_bytecode/java_object_factory.cpp

```diff
@@ -1207,6 +1207,29 @@ void java_object_factoryt::array_loop_init_code(
     plus_exprt(array_init_symexpr, counter_expr, array_init_symexpr.type()),
     array_init_symexpr.type().subtype());
 
+  bool new_item_is_primitive = arraycellref.type().id() != ID_pointer;
+
+  // Use a temporary to initialise a new, or update an existing, non-primitive.
+  // This makes it clearer that in a sequence like
+  // `new_item->x = y; new_item->z = w;` that all the `new_item` references must
+  // alias, cf. the harder-to-analyse
+  // `some_expr[idx]->x = y; some_expr[idx]->z = w;`
+  exprt init_expr;
+  if(new_item_is_primitive)
```

Comment:
:bulb: Any reason to not to do the primitives in the same way, just in the interest of keeping this code simpler?

Reply:
Just to keep it simple indeed. `array[idx] = nondet(int)` -> `new_array_item = nondet(int); array[idx] = new_array_item;` seems like a disservice to the reader.